src/lib_c/x86_64-windows-msvc/c/iocp.cr

```diff
@@ -0,0 +1,53 @@
+require "c/winnt"
+
+# WSAOVERLAPPED is the primary communication structure for async I/O on Windows.
+# See https://docs.microsoft.com/en-us/windows/win32/api/winsock2/ns-winsock2-wsaoverlapped
+@[Extern]
+struct WSAOVERLAPPED
+  internal : LibC::ULONG_PTR
+  internalHigh : LibC::ULONG_PTR
+  offset : LibC::DWORD
+  offsetHigh : LibC::DWORD
+  hEvent : LibC::HANDLE
+  property cEvent : Void*
+
+  def initialize(crystal_event : Crystal::Event)
+    @cEvent = crystal_event.unsafe_as(Pointer(Void))
+  end
+end
+
+@[Link("advapi32")]
+lib LibC
+  struct OVERLAPPED_ENTRY
+    lpCompletionKey : ULONG_PTR
+    lpOverlapped : WSAOVERLAPPED*
+    internal : ULONG_PTR
+    dwNumberOfBytesTransferred : DWORD
+  end
+
+  fun WSAGetLastError : Int
+
+  fun CreateIoCompletionPort(
+    fileHandle : HANDLE,
+    existingCompletionPort : HANDLE,
+    completionKey : ULONG_PTR,
+    numberOfConcurrentThreads : DWORD
+  ) : HANDLE
+
+  fun GetQueuedCompletionStatus(
+    completionPort : HANDLE,
+    lpNumberOfBytesTransferred : DWORD*,
+    lpCompletionKey : ULONG_PTR*,
+    lpOverlapped : WSAOVERLAPPED*,
+    dwMilliseconds : DWORD
+  ) : BOOL
+
+  fun GetQueuedCompletionStatusEx(
+    completionPort : HANDLE,
+    lpCompletionPortEntries : OVERLAPPED_ENTRY*,
+    ulCount : ULong,
+    ulNumEntriesRemoved : ULong*,
+    dwMilliseconds : DWORD,
+    fAlertable : BOOL
+  ) : BOOL
+end
```

Comment:
Should the definitions in this `iocp.cr` move to other places? That's because almost definitions in this directory are placed in the `.cr` files corresponding to `.h` files. For examples, `CreateFileW` is in [`fileapi.cr`](https://github.com/crystal-lang/crystal/blob/0.35.1/src/lib_c/x86_64-windows-msvc/c/fileapi.cr#L42-L44) because it is declared in [`fileapi.h`](https://docs.microsoft.com/en-us/windows/win32/api/fileapi/). `_wmkdir` is in [`dirent.cr`](https://github.com/crystal-lang/crystal/blob/0.35.1/src/lib_c/x86_64-windows-msvc/c/direct.cr#L4) because it is declared [`dirent.h`](https://docs.microsoft.com/en-us/cpp/c-runtime-library/reference/mkdir-wmkdir?view=msvc-160#requirements). Well, I think this is just a custom and not documented. Not all definitions comply. For example `OVERLAPPED` is in `fileapi.cr` though it is declared in [`minwinbase.h`](https://docs.microsoft.com/en-us/windows/win32/api/minwinbase/).

I think that it is better to move them as follows:
`OVERLAPPED_ENTRY` (along with `OVERLAPPED`?) ->  `minwinbase.cr`
`CreateIoCompletionPort`, `GetQueuedCompletionStatus`, `GetQueuedCompletionStatusEx` -> `ioapiset.cr`